pkg/artifactory/resource_artifactory_remote_repository.go

```diff
@@ -474,6 +481,10 @@ func packRemoteRepo(repo *v1.RemoteRepository, d *schema.ResourceData) error {
 		logErr(d.Set("password", getMD5Hash(*repo.Password)))
 	}
 
+	if *repo.PackageType != "generic" && repo.PropagateQueryParams != nil {
+		return fmt.Errorf("Cannot use propagate_query_params with repository type %s. This parameter can be used only with generic repositories", repo.PackageType)
```

Comment:
do you happen to have one artifactory testing licence that I could use to run the tests ?

Reply:
please messsaging me in gitter - I can't give you a license but I can give you access to a server to test

Reply:
New tests pass using the instance.
Some tests are failing using HEAD of master branch:

```
$ go test -v ./pkg/... -run TestAccRemoteRepository_                                                                                                                                                                                      
=== RUN   TestAccRemoteRepository_basic
--- PASS: TestAccRemoteRepository_basic (16.10s)
=== RUN   TestAccRemoteRepository_nugetNew
    testing.go:654: Step 0 error: errors during apply:

        Error: PUT https://jfroche-rt.jfrog.tech/artifactory/api/repositories/terraform-remote-test-repo-nuget: 400 [{Status:400 Message:NuGet Repository configuration is missing mandatory field downloadContextPath
        }]

          on /run/user/1001/tf-test420037412/main.tf line 2:
          (source code not available)


    testing.go:715: Error destroying resource! WARNING: Dangling resources
        may exist. The full state and error is shown below.

        Error: Check failed: not found artifactory_remote_repository.terraform-remote-test-repo-nuget

        State: <no state>
--- FAIL: TestAccRemoteRepository_nugetNew (5.60s)
=== RUN   TestAccRemoteRepository_full
    testing.go:654: Step 0 error: Check failed: Check 5/36 error: artifactory_remote_repository.terraform-remote-test-repo-full: Attribute 'password' expected "187f4facbba536424371d66b531b2d07bed7fe3443b2d84dd794f5f5d7dbca9d", got "b2490f8c35a3da10c58873901a42d7d61493afdcda7a1384f40a73208956ebe7"
--- FAIL: TestAccRemoteRepository_full (10.66s)
FAIL
FAIL    github.com/atlassian/terraform-provider-artifactory/pkg/artifactory     32.362s
FAIL
```
do you face the same ?

Reply:
Yes. Sadly - it's a well know issue that we have numerous tests that don't pass. 

If you're willing to do it: Isolate the tests that don't pass and disable them. You can even commit this code.
Run your changes and test what's left. 

It's high on our priority list to get these tests fixed so that enthusiastic contributors like you don't have these sorts of frustrations. I am really sorry

Reply:
Fixed tests for remote repositories